src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let res = '';
+
+    for (const value of this) {
+      if (value !== null && value !== undefined) {
+        res += value;
+      }
+
+      if (this.indexOf(value) < this.length - 1) {
+        res += separator;
+      }
+    }
+
+    return res;
   };
```

Comment:
use common `for` loop to avoid indexOf which increase complexity of your alghorithm